### PR TITLE
Pin runtime and code-generator to 0.27.1

### PR DESCRIPTION
**Pull request by nnbu**
Description of changes:

1. Update go.mod. Build using latest code-generator.
2. Add dummy test/e2e/service_bootstrap.py and test/e2e/service_cleanup.py to make e2e test happy
3. Add config/iam/recommended-inline-policy

No functionality change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by nnbu** [comment]
@a-hilaly 
Could you please review? There is no functionality change. Only repo update with code-generator.

File: go.mod [hunk 1]
```diff
@@ -3,14 +3,14 @@ module github.com/aws-controllers-k8s/networkfirewall-controller
 go 1.17
```
[the remark this answers is not in the document]
Comment (a-hilaly) [reply go.mod:3]: @acornett21  Ideally yeah - the recent patch we sent to test-infra will ignore those and use the same version to build the controllers.

**Review by a-hilaly — APPROVED** [review]
👍 
thank you @nnbu !

File: test/e2e/service_cleanup.py
Comment (a-hilaly) [inline test/e2e/service_cleanup.py:1]: I think we're missing a requirements.txt file

**Comment by nnbu** [comment]
@a-hilaly How can I make e2e pass? There are no tests running actually. But still it fails with `2023-12-14T18:59:30 [ERROR] Tests failed with exit code 5`

**Comment by a-hilaly** [comment]
@nnbu merging manually as this is not an error.

**Pull request merged**